classes/Twig/Block/ChannelInfoBlock.php

```diff
@@ -79,7 +84,130 @@ public function render()
             '@ModuleAutoUpgrade/block/channelInfo.twig',
             [
                 'upgradeInfo' => $upgradeInfo,
+                'allPhpVersions' => UpgradeSelfCheck::PHP_VERSIONS_DISPLAY,
+                'psPhpCompatibilityRanges' => $this->buildCompatibilityTableDisplay(),
+                'currentFormattedPhpVersion' => $this->getFormattedVersion(PHP_VERSION),
+                'targetFormattedPSVersion' => $this->getFormattedVersion($upgradeInfo['version_num'], self::PS_VERSION_DISPLAY_MAX_PRECISION),
             ]
         );
     }
+
+    /**
+     * Builds array of formatted data for the compatibility table display
+     *
+     * @return array
+     */
+    public function buildCompatibilityTableDisplay()
+    {
+        $startPrestaShopVersion = $previousPHPRange = $previousPrestaVersion = null;
+        $result = [];
+        $i = 0;
+        foreach (UpgradeSelfCheck::PHP_PS_VERSIONS as $prestashopVersion => $phpVersions) {
+            ++$i;
+            if (is_null($startPrestaShopVersion)) {
+                $startPrestaShopVersion = $prestashopVersion;
+                $previousPHPRange = $phpVersions;
+            }
+
+            $isCurrentPrestaVersion = $this->isCurrentPrestashopVersion($startPrestaShopVersion);
+            if ($phpVersions === $previousPHPRange) {
+                $previousPrestaVersion = $prestashopVersion;
+            } else {
+                $label = $this->buildPSLabel($startPrestaShopVersion, $previousPrestaVersion);
+                $result[$label]['php_versions'] = $this->buildPhpVersionsList($previousPHPRange);
+                $result[$label]['is_current'] = $isCurrentPrestaVersion;
+                $startPrestaShopVersion = $prestashopVersion;
+                $result[$label]['is_target'] = $this->getFormattedVersion($this->channelInfo->getInfo()['version_num'], self::PS_VERSION_DISPLAY_MAX_PRECISION) === $label;
+                $previousPrestaVersion = null;
+            }
+            if ($i === count(UpgradeSelfCheck::PHP_PS_VERSIONS)) {
```

Comment:
```suggestion
            if ($i === $numberOfPhpVersions) {
```

Do the count check outside the loop